example/pickdemo.c

```diff
@@ -0,0 +1,191 @@
+	nvgFill(vg);
+	if (DoPick) nvgFillHitRegion(vg, id++);
+	
+	// Lines
+	
+	nvgBeginPath(vg);
+	nvgLineCap(vg, NVG_ROUND);
+	nvgStrokeWidth(vg, 25.0f);
+	nvgStrokeColor(vg, (pickedID == id) ? nvgRGB(255, 0, 0) : nvgRGB(0, 255, 0) );
+	nvgMoveTo(vg, 300, 50);
+	nvgLineTo(vg, 300, 150);
+	nvgStroke(vg);
+	if (DoPick) nvgStrokeHitRegion(vg, id++);
+
+	nvgBeginPath(vg);
+	nvgLineCap(vg, NVG_BUTT);
+	nvgStrokeWidth(vg, 25.0f);
+	nvgStrokeColor(vg, (pickedID == id) ? nvgRGB(255, 0, 0) : nvgRGB(0, 255, 0) );
+	nvgMoveTo(vg, 400, 50);
+	nvgLineTo(vg, 400, 150);
+	nvgStroke(vg);
+	if (DoPick) nvgStrokeHitRegion(vg, id++);
+
+	nvgBeginPath(vg);
+	nvgLineCap(vg, NVG_SQUARE);
+	nvgStrokeWidth(vg, 25.0f);
+	nvgStrokeColor(vg, (pickedID == id) ? nvgRGB(255, 0, 0) : nvgRGB(0, 255, 0) );
+	nvgMoveTo(vg, 500, 50);
+	nvgLineTo(vg, 500, 150);
+	nvgStroke(vg);
+	if (DoPick) nvgStrokeHitRegion(vg, id++);
+
+	// Curves
+	nvgBeginPath(vg);
+	nvgLineCap(vg, NVG_ROUND);
+	nvgStrokeWidth(vg, 25.0f);
+	nvgStrokeColor(vg, (pickedID == id) ? nvgRGB(255, 0, 0) : nvgRGB(0, 255, 0) );
+	nvgMoveTo(vg, 300, 300);
+	nvgBezierTo(vg, 200, 350,  400, 450,  300, 500);
+	nvgStroke(vg);
+	if (DoPick) nvgStrokeHitRegion(vg, id++);
+
+	nvgBeginPath(vg);
+	nvgLineCap(vg, NVG_BUTT);
+	nvgStrokeWidth(vg, 25.0f);
+	nvgStrokeColor(vg, (pickedID == id) ? nvgRGB(255, 0, 0) : nvgRGB(0, 255, 0) );
+	nvgMoveTo(vg, 400, 300);
+	nvgBezierTo(vg, 300, 350,  500, 450,  400, 500);
+	nvgStroke(vg);
+	if (DoPick) nvgStrokeHitRegion(vg, id++);
+
+	nvgBeginPath(vg);
+	nvgLineCap(vg, NVG_SQUARE);
+	nvgStrokeWidth(vg, 25.0f);
+	nvgStrokeColor(vg, (pickedID == id) ? nvgRGB(255, 0, 0) : nvgRGB(0, 255, 0) );
+	nvgMoveTo(vg, 500, 300);
+	nvgBezierTo(vg, 400, 350,  600, 450,  500, 500);
+	nvgStroke(vg);
+	if (DoPick) nvgStrokeHitRegion(vg, id++);
+
+	// Curve Fill
+	nvgBeginPath(vg);
+	nvgMoveTo(vg, 600, 100);
+	nvgBezierTo(vg, 650, 0,  750, 300,  800, 100);
+	nvgLineTo(vg, 800, 250);
+	nvgLineTo(vg, 600, 250);
+	nvgClosePath(vg);
+	nvgFillColor(vg, (pickedID == id) ? nvgRGB(255, 0, 0) : nvgRGB(0, 255, 0) );
+	nvgFill(vg);
+	if (DoPick) nvgFillHitRegion(vg, id++);
+
+	// Curve Stroke
+	nvgBeginPath(vg);
+	nvgMoveTo(vg, 600, 350);
+	nvgBezierTo(vg, 650, 250,  750, 550,  800, 350);
+	nvgLineTo(vg, 800, 500);
+	nvgLineTo(vg, 600, 500);
+	nvgClosePath(vg);
+	nvgLineJoin(vg, NVG_BEVEL);
+	nvgStrokeColor(vg, (pickedID == id) ? nvgRGB(255, 0, 0) : nvgRGB(0, 255, 0) );
+	nvgStroke(vg);
+	if (DoPick) nvgStrokeHitRegion(vg, id++);
+
+	// Rect Fill - Alternate pick api
+	nvgBeginPath(vg);
+	nvgRect(vg, 850, 100, 100, 100);
+	nvgFillColor(vg, (nvgInFill(vg, mx, my)) ? nvgRGB(255, 0, 0) : nvgRGB(0, 255, 0) );
+	nvgFill(vg);
+
+	// Rect Stroke - Alternate pick api
+	nvgBeginPath(vg);
+	nvgRect(vg, 850, 250, 100, 100);
+	nvgStrokeWidth(vg, 25.0f);
+	nvgStrokeColor(vg, (nvgInStroke(vg, mx, my)) ? nvgRGB(255, 0, 0) : nvgRGB(0, 255, 0) );
+	nvgStroke(vg);
+
+}
+
+void saveScreenShot(int w, int h, int premult, const char* name)
```

Comment:
Remove if not used.
I'd prefer you to have pickdemo.h, instead of pickybacking on demo.h